packages/google_maps_flutter/test/circle_updates_test.dart

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+import 'fake_maps_controllers.dart';
+
+Set<Circle> _toSet({Circle p1, Circle p2, Circle p3}) {
+  final Set<Circle> res = Set<Circle>.identity();
+  if (p1 != null) {
+    res.add(p1);
+  }
+  if (p2 != null) {
+    res.add(p2);
+  }
+  if (p3 != null) {
+    res.add(p3);
+  }
+  return res;
+}
+
+Widget _mapWithCircles(Set<Circle> circles) {
+  return Directionality(
+    textDirection: TextDirection.ltr,
+    child: GoogleMap(
+      initialCameraPosition: const CameraPosition(target: LatLng(10.0, 15.0)),
+      circles: circles,
+    ),
+  );
+}
+
+void main() {
+  final FakePlatformViewsController fakePlatformViewsController =
+      FakePlatformViewsController();
+
+  setUpAll(() {
+    SystemChannels.platform_views.setMockMethodCallHandler(
+        fakePlatformViewsController.fakePlatformViewsMethodHandler);
+  });
+
+  setUp(() {
+    fakePlatformViewsController.reset();
+  });
+
+  testWidgets('Initializing a circle', (WidgetTester tester) async {
+    final Circle p1 = Circle(circleId: CircleId("circle_1"));
```

Comment:
nit: rename p1 to c1 and so on. We've use p1 for polylines, m1 for markers. Let's be consistent here.

Reply:
Done.